### PR TITLE
Small test runner improvements

### PR DESCRIPTION
**Pull request by mrgriffin**
- Supports `u16` `PARAMETRIZE`s in the non-battle tests.
- Speeds up `CB2_TestRunner` with a `goto` (saves several frames per skipped test which is valuable when using `TESTS=`).
- Uses DACS to set the test state to `CRASH` if an illegal instruction is encountered. This greatly speeds up test failures when branching into memory that does not contain code. Previously we'd have to wait ~60s (headless!) for `mgba-rom-test` to report a `TIMEOUT`. mgba's 0.10.2 release does not support DACS in its HLE BIOS, but it's on `master` and hopefully the next release will support it. When unsupported, DACS is harmless.

[merrp uses DACS in her crash handler](<https://discord.com/channels/442462691542695948/442465020291317760/1179672160088883280>) (with a different ROM offset).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -495,7 +495,7 @@ $(OBJ_DIR)/ld_script_test.ld: $(LD_SCRIPT_TEST) $(LD_SCRIPT_DEPS)
 $(TESTELF): $(OBJ_DIR)/ld_script_test.ld $(OBJS) $(TEST_OBJS) libagbsyscall tools check-tools
 	@echo "cd $(OBJ_DIR) && $(LD) -T ld_script_test.ld -o ../../$@ <objects> <test-objects> <lib>"
 	@cd $(OBJ_DIR) && $(LD) $(TESTLDFLAGS) -T ld_script_test.ld -o ../../$@ $(OBJS_REL) $(TEST_OBJS_REL) $(LIB)
-	$(FIX) $@ -t"$(TITLE)" -c$(GAME_CODE) -m$(MAKER_CODE) -r$(REVISION) --silent
+	$(FIX) $@ -t"$(TITLE)" -c$(GAME_CODE) -m$(MAKER_CODE) -r$(REVISION) -d0 --silent
```
Comment (DizzyEggg) [inline Makefile:498]: What does `-d0` do?

Comment (mrgriffin) [reply Makefile:498]: It enables the 1MBIT DACS, see [GBATEK](<https://problemkaputt.de/gbatek.htm#gbacartridgeheader>), specifically **09Ch Bit 2,7 - Debugging Enable** and ** 0B4h - Device type, 1 Byte**.

File: test/test_runner.c [hunk 2]
```diff
@@ -361,6 +363,9 @@ void CB2_TestRunner(void)
         MgbaExit_(gTestRunnerState.exitCode);
         break;
     }
+
+    if (gMain.callback2 == CB2_TestRunner)
+        goto top;
```
Comment (AsparagusEduardo) [inline test/test_runner.c:368]: image

Comment (mrgriffin) [reply test/test_runner.c:368]: Haha! I was discussing this with jiang, and I think they found that simply doing `gMain.callback2();` at the end didn't work. I can't remember if we tried `if (gMain.callback2 == CB2_TestRunner) CB2_TestRunner();`.

You'd think that either of those should work, provided that GCC does tail-call elimination (if it doesn't, then it'll blow the stack).

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**